packages/geohash_bin/geohash_bin.0.1/opam

```diff
@@ -0,0 +1,42 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis:
+  "#🌐 geohash converter 🐫. Commandline and CGI. Zero-config, single-file"
+description: """
+Convert WGS84 lat/lon pairs to [Gustavo Niemeyer](http://niemeyer.net/)s
+[Geohash](http://en.wikipedia.org/wiki/Geohash) and back. Web and commandline, 🐪,
+statically linked, single-file, zero-config."""
+maintainer: ["Marcus Rohrmoser <work@mro.name>"]
+authors: ["The GeoHash# programmers"]
+license: "GPL-3.0-only"
+tags: ["Social" "Web" "Geohash" "CGI" "RFC3875"]
+homepage: "https://demo.mro.name/geohash.cgi"
+bug-reports: "https://codeberg.org/mro/geohash/issues"
+depends: [
+  "dune" {>= "3.0"}
```

Comment:
```suggestion
  "ocaml" {>= "4.08"}
  "dune" {>= "3.0"}
```
Due to
```

#=== ERROR while compiling geohash_bin.0.1 ====================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.03.0 | pinned(https://codeberg.org/attachments/9d309a17-96c0-49e4-a936-3331ebf09979)
# path                 ~/.opam/4.03/.opam-switch/build/geohash_bin.0.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p geohash_bin -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/geohash_bin-7-fded00.env
# output-file          ~/.opam/log/geohash_bin-7-fded00.out
### output ###
# (cd _build/default && /home/opam/.opam/4.03/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.lib.objs/byte -I /home/opam/.opam/4.03/lib/re -I /home/opam/.opam/4.03/lib/result -I /home/opam/.opam/4.03/lib/seq -I /home/opam/.opam/4.03/lib/tyre -no-alias-deps -open Lib -o lib/.lib.objs/byte/lib__Cgi.cmo -c -impl lib/cgi.ml)
# File "lib/cgi.ml", line 72, characters 2-13:
# Error: Unbound value Result.bind
```

Reply:
the `ocaml-base-compiler.4.03.0` isn't what I intended adding `"ocaml" {>= "4.08"}`

How can I specify the minimum ocaml version?

Reply:
oh, hold, it isn't yet but has to be?

Reply:
strange, the `geohash.opam` has a `"ocaml" {>= "4.05"}` - where has it to go instead?

Reply:
My suggestion is exactly how you specify a lower bound on the installed ocaml version.

The `Result` module was added in ocaml 4.08 and is used in `geohash_bin`, so you need to add the ocaml dependency with that lower bound on the `geohash_bin` package as per the suggestion above.

Reply:
In your commit 351b614 you need 4.08 as a bound for the Result module, not 4.05

Reply:
I am a bit confused - I just unearthed an old 4.05.0+musl+static+flambda switch and got a failure complaining about a missing musl-clang. Sure, there always has been musl-gcc. https://packages.ubuntu.com/jammy/musl-tools

Reply:
ok, I see.

Reply:
The 4.03 of the error message confused me because it is < 4.05.

Reply:
I don't know what you are trying to do and why you want that special configuration. If you just want to test ocaml 4.05 I would use the base compiler. Static compilation was broken with old compilers if I recall correctly.

Reply:
immediately before reading your comment saying 4.08 is the minimum for Result, I wanted to check against 4.05 if it compiles.

As the version in the error message (4.03) didn't match the one from the opam file (4.05), I wasn't sure about your 4.08 either. So I went with what was there first in the first place and wanted to compile and check with that.

Now I added your initial suggesting and ignore the 4.03 in the CI-toolchain error message.

Thanks for bearing with my first opam package.